src/components/engine/create-engine-instance.tsx

```diff
@@ -39,12 +45,14 @@ export const CreateEngineInstanceButton = ({
   const trackEvent = useTrack();
   const toast = useToast();
   const accountQuery = useAccount();
+  const [engineTier, setEngineTier] = useState<EngineTier>("STARTER");
 
-  const requestCloudHostedEngine = async () => {
+  const addCloudHostedEngine = async () => {
     trackEvent({
       category: "engine",
       action: "click",
       label: "clicked-cloud-hosted",
+      tier: engineTier,
```

Comment:
and then everywhere below *prefer* the explicitly passed value over the state value

Reply:
I made to pass `tier ?? engineTier` to be consistent with what is deployed